File: PHP/MaximumMatchingOfPlayersWithTrainersTest.php
<?php
/**
 * @version PHP 8.2.22
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MaximumMatchingOfPlayersWithTrainersTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MaximumMatchingOfPlayersWithTrainers.php");

class MaximumMatchingOfPlayersWithTrainersTest extends TestCase {

    public function testMaximumMatchingOfPlayersWithTrainers() {
        $obj = new MaximumMatchingOfPlayersWithTrainers();
        //Test 1
        $want = 2;
        $got = $obj->matchPlayersAndTrainers( [4,7,9], [8,2,5,8] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 1;
        $got = $obj->matchPlayersAndTrainers( [1,1,1], [10] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MaximumMatchingOfPlayersWithTrainers_test.go
/**
 * @version GO 1.23.5
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMaximumMatchingOfPlayersWithTrainers
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMaximumMatchingOfPlayersWithTrainers(t *testing.T) {
	tables := []struct {
		want int
		s    []int
		k    []int
	}{
		{2, []int{4, 7, 9}, []int{8, 2, 5, 8}},
		{1, []int{1, 1, 1}, []int{10}},
	}

	for _, table := range tables {
		got := matchPlayersAndTrainers(table.s, table.k)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
